src/probes/journal.c

```diff
@@ -151,17 +161,11 @@ static bool process_existing_entries(sd_journal *journal)
         ret = read_new_entries(journal);
         if (ret < 0) {
                 return false;
-        }
-
-        if (!payload) {
+        } else if (ret == 0) {
```

Comment:
This is an excellent observation. I wasn't considering how the probe should behave in the event of `send_data` failures. I suppose I could call `sd_journal_previous` to move the pointer in the reverse direction in this case, but a timeout would be needed...